File: src/main/java/com/converter/api/service/AbstractUserService.java
package com.converter.api.service;

import com.converter.api.model.User;
import com.converter.api.repository.UserRepository;
import org.springframework.stereotype.Service;

@Service
public abstract class AbstractUserService implements UserService{

    protected final UserRepository userRepository;

    public AbstractUserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Override
    public User create(User user) {
        return userRepository.save(user);
    }
}

File: src/main/java/com/converter/api/service/AbstractUserService.kt
package com.converter.api.service

import com.converter.api.model.User
import com.converter.api.repository.UserRepository
import com.converter.api.service.UserService
import org.springframework.stereotype.Service

@Service
abstract class AbstractUserService(
    protected val userRepository: UserRepository) : UserService {

    override fun create(user: User): User {
        return userRepository.save(user)
    }
}